nav2_behaviors/plugins/drive_on_heading.hpp

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Joshua Wallace
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIORS__PLUGINS__DRIVE_ON_HEADING_HPP_
+#define NAV2_BEHAVIORS__PLUGINS__DRIVE_ON_HEADING_HPP_
+
+#include <chrono>
+#include <memory>
+
+#include "nav2_behaviors/timed_behavior.hpp"
+#include "nav2_msgs/action/drive_on_heading.hpp"
+
+namespace nav2_behaviors
+{
+using DriveOnHeadingAction = nav2_msgs::action::DriveOnHeading;
+
+/**
+ * @class nav2_behaviors::DriveOnHeading
+ * @brief An action server Behavior for spinning in
+ */
+template<typename ActionT>
```

Comment:
You can set `<typename ActionT = nav2_msgs::action::DriveOnHeading>` and then if no template is defined, it should default to that so you don't need to specify in the plugin registration / xml

Reply:
but test, I might be wrong on the syntax